tests/infra/remote.py

```diff
@@ -60,15 +61,22 @@ def log_errors(out_path, err_path):
     error_filter = ["fail", "fatal"]
     try:
         errors = 0
+        tail_lines = deque(maxlen=15)
         with open(out_path, "r") as lines:
             for line in lines:
-                if any(x in line for x in error_filter):
-                    LOG.error("{}: {}".format(out_path, line.rstrip()))
+                stripped_line = line.rstrip()
+                tail_lines.append(stripped_line)
+                if any(x in stripped_line for x in error_filter):
+                    LOG.error("{}: {}".format(out_path, stripped_line))
                     errors += 1
+        if errors:
```

Comment:
Should this branch be merged with the `if errors:` on line 76?

Reply:
Yes, done